app/components/Modals/TokenModal/TokenModal.jsx

```diff
@@ -90,28 +92,48 @@ class TokenModal extends Component<Props, State> {
     })
   }
 
+  shouldDisableSaveButton = () => {
+    const { tokens } = this.state
+
+    const invalidUserGeneratedTokens = tokens
+      .filter((token: TokenItemType) => token.isUserGenerated)
+      .filter((token: TokenItemType) => !token.scriptHash.length)
+
+    if (tokens === this.props.tokens) return true
+    if (invalidUserGeneratedTokens.length) return true
+    if (tokens.length === this.props.tokens.length) return true
```

Comment:
since all three conditions return `true` then you can write them in the same condition
if (tokens === this.props.tokens || invalidUserGeneratedTokens.length || tokens.length === this.props.tokens.length) { ... }

also, I assume `tokens` is an object? if so, better use lodash `isEqual`

Reply:
👍